packages/core/logger/src/Logger.js

```diff
@@ -74,29 +75,46 @@ export function patchConsole() {
   /* eslint-disable no-console */
   // $FlowFixMe
   console.log = console.info = (...messages: Array<mixed>) => {
-    logger.info(joinLogMessages(messages));
+    logger.info(messagesToDiagnostic(messages));
   };
 
   // $FlowFixMe
   console.debug = (...messages: Array<mixed>) => {
     // TODO: dedicated debug level?
-    logger.verbose(joinLogMessages(messages));
+    logger.verbose(messagesToDiagnostic(messages));
   };
 
   // $FlowFixMe
   console.warn = (...messages: Array<mixed>) => {
-    logger.warn(joinLogMessages(messages));
+    logger.warn(messagesToDiagnostic(messages));
   };
 
   // $FlowFixMe
   console.error = (...messages: Array<mixed>) => {
-    logger.error(joinLogMessages(messages));
+    logger.error(messagesToDiagnostic(messages));
   };
 
   /* eslint-enable no-console */
   consolePatched = true;
 }
 
+function messagesToDiagnostic(messages: Array<mixed>): Diagnostic {
+  if (messages.length === 1 && messages[0] instanceof Error) {
+    let error: Error = messages[0];
+
+    return {
+      message: error.message,
+      origin: '@parcel/logger',
```

Comment:
should the default origin be `@parcel/logger`. Might be confusing as the error did not originate from the logger.

Perhaps it should be `unknown` or just an empty string?

Reply:
Changed it to `console` and `Error`